File: backend/src/models/bridge.rs
// src/models/bridge.rs - Bridge operation data models (Phase 4 placeholder)
use serde::{Deserialize, Serialize};
use validator::Validate;
use uuid::Uuid;
use chrono::{DateTime, Utc};

#[derive(Debug, Serialize, Deserialize, Validate)]
pub struct BridgeSwapRequest {
    #[validate(length(min = 1, max = 20))]
    pub source_chain: String,

    #[validate(length(min = 1, max = 20))]
    pub destination_chain: String,

    #[validate(length(min = 1, max = 10))]
    pub source_token: String,

    #[validate(length(min = 1, max = 10))]
    pub destination_token: String,

    #[validate(range(min = 0.000001))]
    pub amount: f64,

    pub quantum_key_id: Option<Uuid>,

    #[validate(range(min = 1, max = 168))] // 1 hour to 1 week
    pub expires_in_hours: Option<u32>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct BridgeSwapResponse {
    pub transaction_id: Uuid,
    pub status: TransactionStatus,
    pub estimated_completion: DateTime<Utc>,
    pub bridge_fee: f64,
    pub network_fee: f64,
    pub quantum_protection_fee: f64,
    pub exchange_rate: f64,
    pub expected_output: f64,
    pub risk_score: f64,
    pub quantum_protected: bool,
}

#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
pub enum TransactionStatus {
    Pending,
    Validating,
    Locked,
    Processing,
    Confirming,
    Confirmed,
    Completed,
    Failed,
    Cancelled,
    Expired,
    Refunded,
}

File: frontend/src/types/bridge.ts
export type ChainType = "ethereum" | "near";

export interface BridgeToken {
  symbol: string;
  name: string;
  address: string;
  decimals: number;
  logoUrl?: string;
  chain: ChainType;
  balance?: string;
  usdValue?: string;
}

export interface BridgeQuoteParams {
  fromToken: string;
  toToken: string;
  fromChain: ChainType;
  toChain: ChainType;
  fromAmount: string;
  maxSlippage?: number;
  userId?: string;
}

export interface BridgeQuote {
  id: string;
  fromToken: string;
  toToken: string;
  fromChain: ChainType;
  toChain: ChainType;
  fromAmount: string;
  toAmount: string;
  exchangeRate: string;
  estimatedGas: string;
  bridgeFee: string;
  protocolFee: string;
  totalFees: string;
  priceImpact: string;
  slippage: string;
  estimatedTime: number; // seconds
  expiresAt: string;
  quantumProtected: boolean;
  riskScore?: number;
}

export interface SwapFormData {
  fromToken: BridgeToken;
  toToken: BridgeToken;
  fromChain: ChainType;
  toChain: ChainType;
  amount: string;
  slippage: number;
  recipient?: string;
}

export interface BridgeSwapRequest {
  quoteId: string;
  fromChain: ChainType;
  toChain: ChainType;
  fromToken: string;
  toToken: string;
  amount: string;
  recipient: string;
  maxSlippage: number;
}

export interface BridgeSwapResponse {
  transactionId: string;
  status: TransactionStatus;
  fromTxHash?: string;
  toTxHash?: string;
  estimatedCompletion: string;
  quantumKeyId?: string;
}

export type TransactionStatus =
  | "pending"
  | "in_progress"
  | "validating"
  | "locked"
  | "processing"
  | "confirming"
  | "confirmed"
  | "completed"
  | "failed"
  | "cancelled"
  | "expired"
  | "refunded";

export interface TransactionProgress {
  transactionId: string;
  status: TransactionStatus;
  progress: number; // 0-100
  currentStep: string;
  steps: TransactionStep[];
  fromTxHash?: string;
  toTxHash?: string;
  errorMessage?: string;
  estimatedTimeRemaining?: number;
}

export interface TransactionStep {
  name: string;
  status: "pending" | "in_progress" | "completed" | "failed";
  description: string;
  txHash?: string;
  timestamp?: string;
}

export interface TransactionHistoryItem {
  id: string;
  fromChain: ChainType;
  toChain: ChainType;
  fromToken: string;
  toToken: string;
  fromAmount: string;
  toAmount: string;
  status: TransactionStatus;
  createdAt: string;
  completedAt?: string;
  fromTxHash?: string;
  toTxHash?: string;
  usdValue?: string;
}

// API Response types
export interface BridgeQuoteResponse {
  success: boolean;
  data?: BridgeQuote;
  error?: string;
}

export interface BridgeSwapResponseData {
  success: boolean;
  data?: BridgeSwapResponse;
  error?: string;
}

export interface BridgeHistoryResponse {
  success: boolean;
  data?: {
    transactions: TransactionHistoryItem[];
    total: number;
    page: number;
    pageSize: number;
  };
  error?: string;
}
